### PR TITLE
feat(components): add a Badge component

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* add a Badge component

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/9320

## Screenshots:

See Storybook

### CODE REVIEW

**Comment by mroz22** [comment]
rebase on this to fix type-check https://github.com/trezor/trezor-suite/pull/9327

File: packages/components/src/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    variant = 'neutral',
+    isDisabled,
+    icon,
+    hasAlert,
+    onClick,
+    children,
+}: BadgeProps) => {
+    const IconComponent = icon ? (
+        <Icon name={icon} color={getIconColor(variant, isDisabled)} />
+    ) : null;
```
Comment (komret) [inline packages/components/src/components/Badge/Badge.tsx]: nit: could be inside other JSX

File: packages/components/src/components/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        size === 'small' ? `0 ${spacingsPx.xs}` : `${spacingsPx.xxxs} ${spacingsPx.xs}`};
+    border-radius: ${borders.radii.full};
+    border: 1px solid transparent;
+    background: ${({ variant, isDisabled, theme }) =>
+        getBackgroundColor(variant, isDisabled, theme)};
+
+    ${focusShadowStyle}
+
+    ${({ onClick }) =>
+        onClick &&
```
Comment (komret) [inline packages/components/src/components/Badge/Badge.tsx:81]:
```suggestion
        onClick && !isDisabled &&
```

**Review by komret — CHANGES_REQUESTED** [review]

File: packages/components/src/components/Badge/Badge.tsx (hunk 1 above)
Comment (dahaca) [reply packages/components/src/components/Badge/Badge.tsx]: aec732c43b

File: packages/components/src/components/Badge/Badge.tsx (hunk 2 above)
Comment (dahaca) [reply packages/components/src/components/Badge/Badge.tsx:81]: aec732c43b

**Comment by komret** [comment]
Still needs rebasing, some unrelated commits got to the PR.

**Review by komret — APPROVED** [review]

**Comment by komret** [comment]
Check validation check.

**Pull request merged**